docs/advanced-topics/optimizations.md

```diff
@@ -34,44 +34,90 @@ impl Example {
-
-This optimization is most useful for data types that aren't `Copy`. If you can copy your data easily, then it probably isn't worth putting it behind a smart pointer. For structures that can contain lots of data like `Vec`, `HashMap`, and `String`, this optimization should be worthwhile.
-
-This optimization works best if the values are never updated by the children, and even better, if they are rarely updated by parents. This makes `Rc<_>s` a good choice for wrapping property values in for pure components.
-
-## View Functions
-
-For code readability reasons, it often makes sense to migrate sections of `html!` to their own functions so you can avoid the rightward drift present in deeply nested HTML.
-
-## Pure Components/Function Components
-
-Pure components are components that don't mutate their state, only displaying content and propagating messages up to normal, mutable components. They differ from view functions in that they can be used from within the `html!` macro using the component syntax \(`<SomePureComponent />`\) instead of expression syntax \(`{some_view_function()}`\), and that depending on their implementation, they can be memoized - preventing re-renders for identical props using the aforementioned `neq_assign` logic.
+## Using smart pointers effectively
+
+**Note: if you're unsure about some of the terms used in this section, the Rust Book has a useful
+chapter which you can find at the bottom of this page (in the "further reading" section).** 
+
+In an effort to avoid cloning large amounts of data to create props when re-rendering, we can use 
+smart pointers to only clone a reference to the data instead of the data itself. If you pass 
+references to the relevant data in your props and child components instead of the actual data you 
+can avoid cloning any data until you need to modify the data in the child component, where you 
+use `Rc::make_mut` to clone and get a mutable reference to the data you want to alter. By avoiding
+cloning data until absolutely necessary, child components can choose not to re-render when supplied
+with props identical to the local copies of props they possess  in `Component::change` calls while 
+incurring limited performance costs. The benefits are especially great when compared to a situation
+in which smart pointers are not used which would mean that the data itself would have to be copied 
+(so that child components could have ownership of the data) before passing the data from parent to 
+child; it would then have to further be compared against the child's local copy of the data. 
```

Comment:
Still not a fan of using "local copy".

Isn't this and the previous part about comparing props just a long-winded way of saying that comparing references is cheaper than comparing the referenced value?
If that's the case I think it should be put more succinctly.

It should also be said that one needs to use `Rc::ptr_eq` to compare rcs by reference. `Rc`'s `PartialEq` implementation compares the inner values for equality. Only if the inner type is marked as `Eq` will it use a pointer equality check first, but if the pointers don't match it will still fall back to comparing the inner values.